examples/tutorials/lighting/app.ts

```diff
@@ -67,31 +53,45 @@ uniform appUniforms {
 out vec4 fragColor;
 
 void main(void) {
-  vec3 materialColor = texture2D(uTexture, vec2(vUV.x, 1.0 - vUV.y)).rgb;
-  vec3 surfaceColor = lighting_getLightColor(materialColor, uApp.eyePosition, vPosition, normalize(vNormal));
+  vec3 surfaceColor = texture2D(uTexture, vec2(vUV.x, 1.0 - vUV.y)).rgb;
+  surfaceColor = lighting_getLightColor(surfaceColor, uApp.eyePosition, vPosition, normalize(vNormal));
 
   fragColor = vec4(surfaceColor, 1.0);
 }
 `;
 
+type AppUniforms = {
+  modelMatrix: NumberArray;
+  mvpMatrix: NumberArray;
+  eyePosition: NumberArray;
+};
+
+const app: ShaderModule<AppUniforms, AppUniforms> = {
+  name: 'app',
+  uniformTypes: {
+    modelMatrix: 'mat4x4<f32>',
+    mvpMatrix: 'mat4x4<f32>',
+    eyePosition: 'vec3<f32>'
+  }
+};
+
+
 // APPLICATION
 
 const eyePosition = [0, 0, 5];
 
 export default class AppAnimationLoopTemplate extends AnimationLoopTemplate {
   static info = INFO_HTML;
 
-  uniformStore = new UniformStore<{
-    app: AppUniforms;
-    lighting: typeof lighting.uniforms;
-    phongMaterial: typeof phongMaterial.uniforms;
-  }>({
-    app,
-    lighting,
-    phongMaterial
-  });
-
   model: Model;
+
+  shaderInputs = new _ShaderInputs<{
```

Comment:
I thought on the API, for the `ShaderInputs` we are passing an object which assigns names (keys) to each registered `ShaderModule`, that are later used when setting the props. 

This is in contrast to the `Model.modules` prop which takes an Array of `ShaderModules`.

When a `Model` is constructed without explicit `shaderInputs` the module name is used for the key, which means the app needs to know this, and isn't aided by types. Is it worth considering changing the `modules` prop to also accept an object so types of the keys can be passed through?

Reply:
I was considering adding such typings to model but I feel the heavy type syntax will make an already complex class even more hard to read.

I would then rather create a `TypedModel` subclass that applies the typings and keeps the Model code complexity in check.

As context, I suspect that Model will be a constant target for feature additions and managing complexity here is important.